lib/kubernetes-deploy/kubernetes_resource/persistent_volume_claim.rb

```diff
@@ -3,16 +3,76 @@ module KubernetesDeploy
   class PersistentVolumeClaim < KubernetesResource
     TIMEOUT = 5.minutes
 
+    def sync(cache)
+      super
+      @storage_classes = cache.get_all("StorageClass").map { |sc| StorageClass.new(sc) }
+    end
+
     def status
       exists? ? @instance_data["status"]["phase"] : "Not Found"
     end
 
     def deploy_succeeded?
-      status == "Bound"
+      return false if deploy_failed?
+      return true if status == "Bound"
+
+      # if the StorageClass has volumeBindingMode: WaitForFirstConsumer,
+      # it won't bind until after a Pod mounts it. But it must be pre-deployed,
+      # as the Pod requires it. So 'Pending' must be treated as a 'Success' state
+      if storage_class&.volume_binding_mode == "WaitForFirstConsumer"
+        status == "Pending" || status == "Bound"
+      end
     end
 
     def deploy_failed?
-      status == "Lost"
+      status == "Lost" || failure_message.present?
+    end
+
+    def failure_message
+      if storage_class_name.nil? && @storage_classes.count(&:default?) > 1
+        "PVC has no StorageClass specified and there are multiple StorageClasses" \
+        "with annoated as default. This is an invalid cluster configuration."
+      end
+    end
+
+    def timeout_message
+      return STANDARD_TIMEOUT_MESSAGE unless storage_class_name.present? && !storage_class
+      "PVC specified a StorageClass of #{storage_class_name} but the resource does not exist"
+    end
+
+    private
+
+    def storage_class_name
+      @definition.dig("spec", "storageClassName")
+    end
+
+    def storage_class
+      if storage_class_name.present?
+        @storage_classes.detect { |sc| sc.name == storage_class_name }
+      elsif storage_class_name != ""
```

Comment:
why not just a bare `else`?

Reply:
`storage_class_name == ""` is actually a different case than `storage_class_name == nil`. The first says I don't want any storage class, the second is I want the default one.

Reply:
I think that's worth a comment in the code

Reply:
A comment and/or explicitly handling that case.

Reply:
done.